packages/fixie/src/auth.ts

```diff
@@ -0,0 +1,207 @@
+import yaml from 'js-yaml';
+import fs from 'fs';
+import terminal from 'terminal-kit';
+import os from 'os';
+import path from 'path';
+import untildify from 'untildify';
+import axios from 'axios';
+import open from 'open';
+import http from 'http';
+import crypto from 'crypto';
+import readline from 'readline';
+import net from 'net';
+
+const { terminal: term } = terminal;
+
+import { FixieClient } from './client.js';
+
+/** Represents contents of the Fixie CLI config file. */
+export interface FixieConfig {
+  apiUrl?: string;
+  apiKey?: string;
+}
+
+export const FIXIE_API_URL = 'https://api.fixie.ai';
+export const FIXIE_CONFIG_FILE = '~/.config/fixie/config.yaml';
+
+/** Load the client configuration from the given file. */
+export function LoadConfig(configFile: string): FixieConfig {
+  const fullPath = untildify(configFile);
+  if (!fs.existsSync(fullPath)) {
+    return {};
+  }
+  const config: object = yaml.load(fs.readFileSync(fullPath, 'utf8')) as object;
+  // Warn if any fields are present in config that are not supported.
+  const validKeys = ['apiUrl', 'apiKey'];
+  const invalidKeys = Object.keys(config).filter((key) => !validKeys.includes(key));
+  for (const key of invalidKeys) {
+    term('❓ Ignoring invalid key ').yellow(key)(` in ${fullPath}\n`);
+  }
+  return config as FixieConfig;
+}
+
+/** Save the client configuration to the given file. */
+export function SaveConfig(config: FixieConfig, configFile: string) {
+  const fullPath = untildify(configFile);
+  const dirName = path.dirname(fullPath);
+  if (!fs.existsSync(dirName)) {
+    fs.mkdirSync(dirName, { recursive: true });
+  }
+  if (fs.existsSync(fullPath)) {
+    // Merge the new config with the existing config, so we don't
+    // overwrite any fields that are not specified.
+    const currentConfig = yaml.load(fs.readFileSync(fullPath, 'utf8')) as object;
```

Comment:
Wouldn't it be better to reuse `loadConfig` for this?

Reply:
Well, I want to retain all the keys in the original config file and loadConfig is intended to only return valid FixieConfig keys.